src/rotator_library/utils/resilient_io.py

```
@@ -677,3 +678,66 @@ def safe_mkdir(path: Union[str, Path], logger: logging.Logger) -> bool:
+            data = self._pending
+            self._pending = None
+
+            try:
+                # Atomic write: temp -> rename
+                temp_path = self.file_path.with_suffix('.tmp')
+                temp_path.write_text(
+                    json.dumps(data, indent=2, default=str),
+                    encoding='utf-8'
+                )
+                temp_path.replace(self.file_path)
+            except Exception:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)